src/ui/components/ExplorerHome/ExplorerHome.js

```diff
@@ -237,6 +242,43 @@ export const ExplorerHome = ({
     [timeScopedCredGrainView]
   );
 
+  const {credTotalsTimeline, grainTotalsTimeline} = useMemo(() => {
+    const allParticipantsCred: Array<Array<number>> = allParticipants.map(
+      (participant) => {
+        return participant.credPerInterval.slice();
+      }
+    );
+    const allParticipantsGrain: Array<Array<Grain>> = allParticipants.map(
+      (participant) => {
+        return participant.grainEarnedPerInterval.slice();
+      }
+    );
+    const credAccumulator = (
+      credTimelineAccumalator: Array<number>,
+      participantCred
+    ) =>
+      credTimelineAccumalator.map(
+        (intervalTotal, i) => intervalTotal + participantCred[i]
+      );
```

Comment:
```suggestion
    const credAccumulator = (
      credTimelineAccumulator: Array<number>,
      participant
    ) =>
      credTimelineAccumulator.map(
        (intervalTotal, i) => intervalTotal + participant.credPerInterval[i]
      );
```
I did read your note about using more variables to increase legibility, however I believe the code would be more legible if you removed lines 245-255 and changed your accumulators like this. And it would be more efficient.

Reply:
Yes, that makes sense, but the current code is the result of lots of fenangling due to the fact that flow hates `.reduce`. We tried your changes and it made the flow monster angryyyyyyyyy - bascially, even though we are using `reduce` to construct a new array, flow complains that the array type of `allParticipants` (`Array<ParticipantCredGrain>`) is incompatible with the desired type of our new array (`Array<number>`).

Reply:
When you are back from your much needed break we could look at this code together to make it more pretty but we are going to leave it for now